nslsii/sync_experiment/sync_experiment.py

```diff
@@ -112,10 +112,16 @@ def sync_experiment(proposal_number, beamline, verbose=False, prefix=""):
         if not should_they_be_here(username, new_data_session, beamline):
             raise AuthorizationError(f"User '{username}' is not allowed to take data on proposal {new_data_session}")
 
+        pi_name = ""
+        for user in users:
+            if user.get("is_pi"):
+                pi_name = " ".join([user.get("first_name", ""), user.get("last_name", "")])
+
         md["data_session"] = new_data_session
         md["username"] = username
         md["start_datetime"] = datetime.now().isoformat()
         md["cycle"] = get_current_cycle()
+        md["proposal"] = {"number": proposal_number, "title": proposal_data.get("title"), "pi_name": pi_name}
```

Comment:
A schema question... Should proposal number be a string (instead of an integer), and maybe called `id` (to match the API output)?

Reply:
It makes sense to me, in case the number is not a number in the future (such as a uid string or something else).

Reply:
Done. Now `['proposal']['id']` is a string.